syntax = "proto3";

import "common.proto";
package autodrive.workflow.proto;

message ImageInfo {
  uint32 width = 1;
  uint32 height = 2;
}

message MemInfo {
  uint64 addr = 1;
  uint32 size = 2;
  uint32 pitch = 3;
}

message ImageMsg {
  // Header
  MsgHeader header = 1;
  // Error
  Error error = 2;
  ImageInfo image_info = 3;
  MemInfo mem_info_y = 4;
  MemInfo mem_info_uv = 5;
}